ExploreASL_ImportMaster.m

```diff
@@ -19,85 +19,86 @@
 
     %% Import Workflow
     
-    if x.opts.bImportData
-        % We expect x.opts.DatasetRoot to be the study root directory, but if it is not defined, 
-        % then the user probably used a path to a descriptive JSON file instead
-        if isfield(x, 'dir') && isfield(x.dir, 'DatasetRoot') && isempty(x.dir.DatasetRoot)
-            x.dir.DatasetRoot = xASL_fileparts(x.opts.DatasetRoot);
-        end
+    % We expect x.opts.DatasetRoot to be the study root directory, but if it is not defined,
+    % then the user probably used a path to a descriptive JSON file instead
+    if isfield(x, 'dir') && isfield(x.dir, 'DatasetRoot') && isempty(x.dir.DatasetRoot)
+        x.dir.DatasetRoot = xASL_fileparts(x.opts.DatasetRoot);
+    end
+    
+    % On default we expect the JSON files to be there
+    missingJSON = false;
 
-        % Check if at least one of the three steps should be performed
-        missingJSON = false; % Fallback
-        if sum(x.opts.ImportModules)>0
-            % DICOM TO NII
-            if x.opts.ImportModules(1)==1
-                if ~isempty(x.dir.sourceStructure)
-                    try
-                        xASL_module_Import(x.dir.DatasetRoot, x.dir.sourceStructure, x.dir.studyPar, [1 0 0 0], false, true, false, x);
-                    catch loggingEntry
-                        ExploreASL_ImportMaster_PrintLoggingEntry('DICOM to NIfTI',loggingEntry);
-                        [x] = xASL_qc_AddLoggingInfo(x, loggingEntry);
-                    end
-                else
-                    missingJSON = true;
-                end
+    % DICOM TO NII
+    if x.opts.ImportModules(1)==1
+        if ~isempty(x.dir.sourceStructure)
+            try
+                xASL_module_Import(x.dir.DatasetRoot, x.dir.sourceStructure, x.dir.studyPar, [1 0 0 0], false, true, false, x);
```

Comment:
Isn't that a bit weird that you call the import module 1 by 1? I mean, I am fine with this strategy, but shouldn't you then rather have 1,2,3,4 instead of [1 0 0 0], [0 1 0 0] etc, when it is never run as a whole?!

Reply:
Because as it is now, I don't see any added value of the ImportMaster script...

Reply:
Hm... I think it got like because of how we used to do things before. We had the testing import scripts where we did it like this and I kind of copied that behavior from there and applied it to the former ExploreASL_Import which is xASL_module_Import now.

Reply:
What you're saying makes sense. The added value of the ImportMaster script is only the try-catch basically.

Reply:
Using 1,2,3,4 doesn't sound that clever to me. We agreed on using binary notation and we already changed the ExploreASL parameters for that. We shouldn't change those things back.

Reply:
@jan-petr: what you were asking for required some bigger changes. I had to change some things for the dataPar creation in derivatives and also some of the error handling. looks pretty good now I think. The ImportMaster is more like a small wrapper now, but still helps with the try-catch and so on. Running the ExploreASL_Master unit test ATM. This is the commit 
7fdaef3  👍

Reply:
@jan-petr good news 👍 

![grafik](https://user-images.githubusercontent.com/46593074/129942882-d855994c-d48e-4681-9b80-a411570d2201.png)